text/0111-combined-dynamic-sampling.md

```diff
@@ -0,0 +1,151 @@
+
+Finally, there is no connection between sampled replays and sampled profiles. A
+trace with a profile does not have a higher chance of having a replay captured
+or vice versa.
+
+# Objective
+
+The goal of this RFC is to increase the affinity of replays and profiles to
+stored (indexed) traces so that the likelihood of capturing traces with **both**
+replays and profiles is maximized.
+
+At the same time, the impact on effective dynamic sampling rates should be
+minimized so that the number of stored events still closely tracks the target
+fidelity.
+
+# Background
+
+Sampling on SDKs and the server is each based on two components:
+
+1. A random number between `0.0` and `1.0`, called "sample seed".
+2. A cutoff percentage, called "sample rate".
+
+The sampler compares the seed to the rate: If the seed is lower, the data is
+sampled, otherwise it is dropped.
+
+Sampling can be performed multiple times in series, either with the same rate or
+with different rates. The outcome of this depends primarily on how the sample
+seed is handled:
+
+- Using different seeds effectively multiplies the sample rates. For example
+  _10% x 5% = 0.5%_. This is what happens between client-side and server-side
+  sampling today.
+- Reusing the same seed repeats the sampling decision. This is how dynamic
+  sampling operates in ingestion without a need for synchronization between
+  transactions or spans. The result is the lowest of all the applied sample
+  rates.
+
+The second option introduced above is also referred to as "coupled sampling",
+since the sample decisions are coupled to each other. It can also be applied to
+correlated data of different kinds each with their own sample rate. This creates
+an interesting effect: Data sampled with the lower sample rate always has the
+other kind of data available, since the decision is coupled.
+
+![coupled-sampling](https://github.com/getsentry/rfcs/assets/1433023/5cf1dca7-b2af-42e2-83ca-231687b12e37)
+
+# Coupled Sampling Decision
+
+This RFC proposes to couple the client-side sample rates of sparse data
+(profiles and replays) between each other and with server-side dynamic sampling:
+
+1. When the trace starts, the SDK computes a sample seed for all sparse data.
+2. Profiles and replays are recorded if their respective sample rates are higher
+   than the sparse data sample seed.
+3. The sparse data sample seed is placed on the DSC and propagated to all
+   connected SDKs, as well as sent in Envelope headers to Sentry.
+4. If available, Sentry uses the sparse data sample seed to compute the
+   server-side dynamic sampling decision instead of the trace ID. If the seed is
+   not present in the DSC, the sever applies dynamic sampling like before.
+
+With this strategy, profiles and replays are sampled in common traces as much as
+the sample rates permit, and dynamic sampling will prefer to store traces with
+all data available.
+
+## Sub-Trace Sampling
```

Comment:
This section will be revisited. Following a conversation with @mydea, we will keep the replay sample rate completely independent of trace sampling (however subject to RFC 109). I'll write up a revision of this section that takes necessary changes into account.